middleware/file/file.go

```diff
@@ -120,6 +121,20 @@ func Parse(f io.Reader, origin, fileName string) (*Zone, error) {
 		if err := z.Insert(x.RR); err != nil {
 			return nil, err
 		}
+
+		if x.RR.Header().Rrtype == dns.TypeDNAME {
+			dnames = append(dnames, x.RR.Header().Name)
+		}
+	}
+	// Resource records MUST NOT exist at any subdomain of the owner of a
+	// DNAME RR. Those names below the DNAME RR will be ignored.
```

Comment:
I think these will be ignored by the lookup, they should be ignore from the database.
AXFR still needs to transfer these and dynamic update may delete the DNAME, exposing the underlaying records.
IOW: just allow these records to exist